pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -538,15 +538,22 @@ func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) (int3
 	// check if Node IP, PSK, or tunnel type changes. This can
 	// happen if removeStaleTunnelPorts fails to remove a "stale"
 	// tunnel port for which the configuration has changed, return error to requeue the Node.
+	hasRemoved := false
 	if ok {
 		if !c.compareInterfaceConfig(interfaceConfig, nodeIP, portName) {
-			return 0, fmt.Errorf("IPSec tunnel interface config doesn't match cached one, stale IPSec tunnel port %s", interfaceConfig.InterfaceName)
-		}
-		klog.V(2).InfoS("Found cached IPSec tunnel interface", "interfaceName", interfaceConfig.InterfaceName, "port", interfaceConfig.OFPort, "nodeName", nodeName)
-		if interfaceConfig.OFPort != 0 {
-			return interfaceConfig.OFPort, nil
+			if err := c.ovsBridgeClient.DeletePort(interfaceConfig.PortUUID); err != nil {
+				return 0, fmt.Errorf("IPSec tunnel interface config doesn't match cached one, fail to delete stale IPSec tunnel port %s: %v", interfaceConfig.InterfaceName, err)
```

Comment:
The first sencence can be a separate log indicating a mismatch is detected and it's going to delete the stale port.
```
klog.InfoS("IPSec tunnel interface config doesn't match the cached one, deleting the stale IPSec tunnel port", "node", nodeName, "interface", interfaceConfig.InterfaceName)
if err := c.ovsBridgeClient.DeletePort(interfaceConfig.PortUUID); err != nil {
       return 0, fmt.Errorf("failed to delete stale IPSec tunnel port %s: %v", interfaceConfig.InterfaceName, err)
}
```

Reply:
done